src/Game.js

```diff
@@ -74,25 +74,42 @@ const playCard = (G, ctx, card) => {
     console.log(`${ctx.playerID} played`);
     const playerId = ctx.playerID;
     const playedCard = G.players[playerId].cards[card];
-    G.players[playerId].cards = G.players[playerId].cards.filter(c => c !== playedCard);
-    G.aboutToPlay[playerId] = null;
-
-    G.cardsPlayed.push(playedCard);
-    G.gameOver = isGameOver(G);
+    return {
+        ...G,
+        players: {
+            [playerId]: {
+                cards: G.players[playerId].cards.filter(c => c !== playedCard)
+            }
+        },
+        aboutToPlay: {
+            [playerId]: null
+        },
+        cardsPlayed: [...G.cardsPlayed, playedCard],
+        gameOver: isGameOver(G)
```

Comment:
oh damn, this might not work. :/ 

I'm pretty sure `isGameOver` needs the _updated_ `G`.

Reply:
Ah, so I think I may have figured out a way to still get it to work. Instead of returning the new object immediately, I decided to make a new object called newG that will act as the copy of G and then modifying newG.gameOver after that object was created. I hope that should solve the problem!